include/deal.II/lac/sparse_matrix_tools.h

```diff
@@ -0,0 +1,440 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_sparse_matrix_tools_h
+#define dealii_sparse_matrix_tools_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/mpi_compute_index_owner_internal.h>
+
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Utilities
+{
+  namespace MPI
+  {
+    template <typename T>
+    std::tuple<T, T>
+    prefix_sum(const T &       value,
+               const MPI_Comm &comm,
+               const bool      exclusive = true)
```

Comment:
@kronbichler Do you thinks such a function would be helpful? We have used prefix sums already at so many places! If yes, I would generalize the function so that it also works for vectors (as we need to process data for all multigrid levels simultaneously).

Reply:
I think that function would make sense, yes.